habitat-lab/habitat/tasks/rearrange/actions/habitat_mas_actions.py

```diff
@@ -321,18 +321,17 @@ def step(self, *args, **kwargs):
             self.pathfinder = self._create_pathfinder(self.config)
 
         self.skill_done = False
-        nav_to_target_idx = kwargs[
+        nav_action = kwargs[
             self._action_arg_prefix + "oracle_nav_action"
         ]
-
-        if self.prev_nav_done and int(nav_to_target_idx[0]) - 1 == self.prev_match_target_id:
+        nav_to_target_idx = int(nav_action[0]) - 1
+        has_nav_action = nav_action[1]
+        if self.prev_nav_done and nav_to_target_idx == self.prev_match_target_id:
             nav_to_target_idx = 0
-        if nav_to_target_idx <= 0 or nav_to_target_idx > len(
+        if nav_to_target_idx < 0 or nav_to_target_idx > len(
```

Comment:
This change might conflict with previous settings. When `self.prev_nav_done` is True, `nav_to_target_idx` is set to 0, nav will continue rather than return.

Reply:
May be we could roll this file to previous version?

Reply:
Solved :)